flask_resty_tenants/authorization.py

```diff
@@ -76,6 +81,19 @@ def get_authorized_tenant_ids(self, role):
 
             yield tenant_id
 
+    def check_request_tenant_id(self):
+        try:
```

Comment:
I am not a big fan of try-except for control flow in general, but try-except-else is even more confusing :P

I feel like this is easier and with less locs:

```py
tenant_id = self.get_request_tenant_id()
if tenant_id is not None and self.get_tenant_role(tenant_id) < self.read_role:
    flask.abort(404)
```

also, checking for `KeyError` ties it to the implementation of `get_request_tenant_id`, so I don't know if it's beneficial to extract that function and make it overrideable at all.